native/src/main/java/io/ballerina/persist/nodegenerator/BalSyntaxTreeGenerator.java

```diff
@@ -0,0 +1,630 @@
+                        BalFileConstants.EMPTY_STRING, BalFileConstants.ERROR)));
+        StringBuilder retRecord = new StringBuilder();
+        create.addStatement(NodeParser.parseStatement(BalFileConstants.CREATE_SQL_RESULTS));
+        if (keys.size() > 1) {
+            for (HashMap map : entity.getFields()) {
+                if (retRecord.length() > 0) {
+                    retRecord.append(", ");
+                }
+                if (map.get(BalFileConstants.AUTOGENERATED).equals(BalFileConstants.TRUE)) {
+                    retRecord.append(String.format(BalFileConstants.RECORD_FIELD_LAST_INSERT_ID,
+                            map.get(BalFileConstants.FIELD_NAME), BalFileConstants.KEYWORD_INT));
+                } else {
+                    retRecord.append(String.format(BalFileConstants.RECORD_FIELD_VALUE,
+                            map.get(BalFileConstants.FIELD_NAME),
+                            map.get(BalFileConstants.FIELD_NAME)));
+                }
+            }
+            if (keyAutoInc) {
+                create.addStatement(NodeParser.parseStatement(String.format(BalFileConstants.RETURN_RECORD_VARIABLE,
+                        retRecord)));
+            } else {
+                create.addStatement(NodeParser.parseStatement(String.format(BalFileConstants.RETURN_VARIABLE,
+                        BalFileConstants.VALUE)));
+            }
+        }  else {
+            for (HashMap map : entity.getFields()) {
+                if (retRecord.length() > 0) {
+                    retRecord.append(", ");
+                }
+                if (map.get(BalFileConstants.FIELD_NAME).equals(entity.getKeys()[0].trim().replaceAll(
+                        BalFileConstants.DOUBLE_QUOTE, BalFileConstants.EMPTY_STRING))) {
+                    retRecord.append(String.format(BalFileConstants.RECORD_FIELD_LAST_INSERT_ID,
+                            map.get(BalFileConstants.FIELD_NAME), keyType));
+                } else {
+                    retRecord.append(String.format(BalFileConstants.RECORD_FIELD_VALUE,
+                            map.get(BalFileConstants.FIELD_NAME),
+                            map.get(BalFileConstants.FIELD_NAME)));
+                }
+            }
+            if (!keyAutoInc) {
+                IfElse valueNilCheck = new IfElse(NodeParser.parseExpression(
+                        BalFileConstants.LAST_RETURN_ID_NULL_CHECK));
+                valueNilCheck.addIfStatement(NodeParser.parseStatement(BalFileConstants.RETURN_VALUE));
+                create.addIfElseStatement(valueNilCheck.getIfElseStatementNode());
+                create.addStatement(NodeParser.parseStatement(String.format(BalFileConstants.RETURN_RECORD_VARIABLE,
+                        retRecord)));
+            } else {
+                create.addStatement(NodeParser.parseStatement(String.format(BalFileConstants.RETURN_RECORD_VARIABLE,
+                        retRecord)));
+            }
```

Comment:
```suggestion
if (!keyAutoInc) {
                IfElse valueNilCheck = new IfElse(NodeParser.parseExpression(
                        BalFileConstants.LAST_RETURN_ID_NULL_CHECK));
                valueNilCheck.addIfStatement(NodeParser.parseStatement(BalFileConstants.RETURN_VALUE));
                create.addIfElseStatement(valueNilCheck.getIfElseStatementNode());
            }
            create.addStatement(NodeParser.parseStatement(String.format(BalFileConstants.RETURN_RECORD_VARIABLE,
                    retRecord)));
```

Reply:
updated in [6d8b4a4](https://github.com/ballerina-platform/persist-tools/pull/15/commits/6d8b4a41b9a0ea1002bb906bb2008380cfa7cdb1)